index.coffee

```diff
@@ -178,10 +187,34 @@ class ResumableRequest extends stream.Readable
 			@error = new Error('Received more bytes than expected!')
 		else if @retries >= @maxRetries
 			@error = new Error('Maximum retries exceeded')
+		else if @bytesTotal? and @bytesRead is @bytesTotal
+			@_end()
+			return
 		else
 			@retries += 1
 			@emit('retry', @progress())
 			@_request()
 			return
 
-		@abort()
+		@_abort(@error)
+
+	abort: ->
+		@request?.abort()
+
+	_abort: (err) ->
+		return if @ended
+		@ended = true
+
+		@response?.resume()
+		@response?.destroy()
+
+		if err?
+			@emit('error', err)
+		@emit('aborted')
+		@emit('end')
```

Comment:
Having a stream emit an "end" event after an error is unusual, normally streams don't emit any more events after an error (there are a few exceptions though iirc)